### ISSUE TITLE
[iOS] getting SKErrorDomain 2

### ISSUE THREAD

**Issue by HarelM** [j3k0/cordova-plugin-purchase#1116]
### system info

Manufacture: Apple
Model: iPhone11,6
Platform: iOS
OS version: 14.0.1
Plugin version: 10.1.1

### Expected behavior
A user had purchased a subscription should be able to use it and see that he purchased
### Observed behavior
The app doesn't get to owned state and calling order returns an error.
### Steps to reproduce
Hard to tell, this doesn't reproduce to most users.
Here are the logs from the app:
```
2020-10-28 09:04:04 | DEBUG | [Store] Approved, verifing: il.org.osm.israelhiking
2020-10-28 09:04:04 | DEBUG | [Store] scheduleValidation()
2020-10-28 09:04:05 | DEBUG | [Store] scheduleValidation()
2020-10-28 09:04:07 | DEBUG | [Store] runValidation()
2020-10-28 09:04:07 | DEBUG | [Store] ajax -> send request to https://validator.fovea.cc/v1/validate?appName=il.org.osm.israelhiking&apiKey=...
2020-10-28 09:04:09 | DEBUG | [Store] validator success, response: ...{"ok":true,"data":{"id":"il.org.osm.israelhiking",...
2020-10-28 09:04:09 | DEBUG | [Store] verify -> {"success":true,"data":{"id":"il.org.osm.israelhiking"...
2020-10-28 09:04:09 | DEBUG | [Store] transaction fields for il.org.osm.israelhiking
2020-10-28 09:04:09 | DEBUG | [Store] verify -> success: {"id":"il.org.osm.israelhiking",...
2020-10-28 09:04:09 | DEBUG | [Store] syncWithAppStoreReceipt
2020-10-28 09:04:09 | DEBUG | [Store] {"type":"ios-appstore",...
2020-10-28 09:04:09 | DEBUG | [Store] transaction fields for offline_map
2020-10-28 09:04:09 | DEBUG | [Store] expiryDate: 2021-03-22T19:06:37.000Z
2020-10-28 09:04:09 | DEBUG | [Store] Verified, Finishing: il.org.osm.israelhiking
2020-10-28 09:04:09 | DEBUG | [Store] product -> defer finishing il.org.osm.israelhiking
2020-10-28 09:04:09 | DEBUG | [Store] transaction fields for il.org.osm.israelhiking
2020-10-28 09:04:09 | DEBUG | [Store] verify -> success: {"id":"il.org.osm.israelhiking"...
2020-10-28 09:04:09 | DEBUG | [Store] transaction fields for il.org.osm.israelhiking
2020-10-28 09:04:09 | DEBUG | [Store] verify -> success: {"id":"il.org.osm.israelhiking"...
2020-10-28 09:04:09 | DEBUG | [Store] ios -> finishing il.org.osm.israelhiking (a application)
2020-10-28 09:04:09 | DEBUG | [Store] product -> finishing il.org.osm.israelhiking
2020-10-28 09:04:09 | DEBUG | [Store] state: il.org.osm.israelhiking -> owned
```
`offline_map` never gets to owned state although it was purchased.
When clicking order we get the following error:
```
2020-10-28 09:04:14 | DEBUG | [Store] Ordering product
2020-10-28 09:04:14 | DEBUG | [Store] state: offline_map -> requested
2020-10-28 09:04:14 | DEBUG | [Store] ios -> is purchasing offline_map
2020-10-28 09:04:14 | DEBUG | [Store] state: offline_map -> initiated
2020-10-28 09:04:18 | ERROR | [Store] ios -> ERROR 6777006: The operation couldn’t be completed. (SKErrorDomain error 2.) - {"productId":"offline_map"}
2020-10-28 09:04:18 | DEBUG | [Store] state: offline_map -> valid
2020-10-28 09:04:18 | ERROR | [Store] error handler: The operation couldn’t be completed. (SKErrorDomain error 2.) (6777006)
```
I've seen something with end transaction here:
https://developer.apple.com/forums/thread/51621
Not sure it is related...
Le me know if there's anything I can do to help...

**Comment by HarelM** [comment]
I've updated the plugin to latest version and I'm still facing this issue.
Any more info I can provide? What else can I test?

**Comment by zstadler** [comment]
@j3k0,

Could you look into this issue?

The customer is unable to use the feature he had purchased.

**Comment by HarelM** [comment]
The customer reported that this issue was solved after I upgraded to version 10.4.0 of this plugin.
I'll close this issue for now, and either open another or ask to reopen this in case the error returns.
Thanks for all your work on this plugin! :-)
